### PR TITLE
Add support for an experimental mode

### LINKED ISSUE #6150: Experimental mode
/kind user-story

## User Story
As a user I want to define ODO_EXPERIMENTAL So that I can use experimental commands and flags

## Acceptance Criteria 

When ODO_EXPERIMENTAL=true
- [ ] It should display in the header the command is running in Experimental mode
- [ ] It should enable experimental commands, which are not visible otherwise
- [ ] It should enable experimental flags, which are not visible otherwise
- [ ] If odo is executed without `ODO_EXPERIMENTAL=true`, experimental commands should appear non-existing (not just hidden) In other words, if `--run-on` is experimental flag `odo --run-on podman` should error out complaining about invalid flag. 

## Links 
- Related Epic (mandatory): #5865

/kind user-story

### PR DESCRIPTION
**Pull request by rm3l**
**What type of PR is this:**
/kind feature

**What does this PR do / why we need it:**
See #6150

**Which issue(s) this PR fixes:**
Fixes #6150

**PR acceptance criteria:**

- [x] Unit test 

- [x] Integration test 

- [x] Documentation
  - Updated: [Configuration](https://deploy-preview-6300--odo-docusaurus-preview.netlify.app/docs/overview/configure#environment-variables-controlling-odo-behavior)
  - New: [Experimental Mode](https://deploy-preview-6300--odo-docusaurus-preview.netlify.app/docs/user-guides/advanced/experimental-mode)

**How to test changes / Special notes to the reviewer:**
You can try it via the generic `--run-on` flag, which has been explicitly marked as experimental in this PR.
So to use it, you must set the `ODO_EXPERIMENTAL_MODE` env var to `true` first. Otherwise, the flag won't be recognized by Cobra and an error will be returned.

### CODE REVIEW

File: pkg/odo/cli/feature/experimental.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+)
+
+var _experimentalFeatures = []OdoFeature{
+	GenericRunOnFlag,
+}
+
+// IsExperimental returns whether the given feature can be used as an experimental CLI feature,
+// which means that the following two conditions should be met: i) the ODO_EXPERIMENTAL_MODE environment variable is enabled,
+// and ii) the provided feature is explicitly listed as experimental in the list of supported experimental features.
+func IsExperimental(feat OdoFeature) bool {
```
[the remark this answers is not in the document]
Comment (rm3l) [reply pkg/odo/cli/feature/experimental.go:23]: > We will also need to cover the case where a feature exists but is not anymore marked as experimental in `_experimentalFeatures`.
> 
> In this case, this feature should be always enabled.
> 
> I don't think this method will cover this case.
> 
> Alternatively, I think a method `IsEnabled` (or similar), that would return True for all non-experimental features (regardless of `OdoExperimentalModeEnvVar` value), and also return True when a feature is marked as Experimental AND `OdoExperimentalModeEnvVar` value is True

Indeed, you're right, this case is not currently covered. I'll make the changes - thanks.

[the remark this answers is not in the document]
Comment (rm3l) [reply pkg/odo/cli/feature/experimental.go]: Changes done in [`aaba7db` (#6300)](https://github.com/redhat-developer/odo/pull/6300/commits/aaba7db00186532d9785b941ae0764f525a6075d).
I took the opportunity to also simplify how a feature is marked as experimental. It was quite redundant to me to have to add it in `_experimentalFeatures`. So now, the `OdoFeature` struct itself has a field that determines whether a feature is experimental or not.

Added an `IsEnabled` method that should work as you suggested. See [`4dc9a6a` (#6300)](https://github.com/redhat-developer/odo/pull/6300/commits/4dc9a6a598b03af1c838eb012f49334f679300d5) for an example with the `--run-on` feature.

**Review by feloy — APPROVED** [review]

**Pull request merged**